ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2

```diff
@@ -484,6 +484,43 @@ interface {{ ethernet_interface.name }}
 {%         if ethernet_interface.pim.ipv4.dr_priority is arista.avd.defined %}
    pim ipv4 dr-priority {{ ethernet_interface.pim.ipv4.dr_priority }}
 {%         endif %}
+{%         if ethernet_interface.poe.priority is arista.avd.defined %}
+   poe priority {{ ethernet_interface.poe.priority }}
+{%         endif %}
+{%         if ethernet_interface.poe.reboot.action is arista.avd.defined %}
+   poe reboot action {{ ethernet_interface.poe.reboot.action }}
+{%         endif %}
+{%         if ethernet_interface.poe.link_down.action is arista.avd.defined %}
+{%             set poe_link_down_action_cli = 'poe link down action ' ~ ethernet_interface.poe.link_down.action %}
+{%             if ethernet_interface.poe.link_down.power_off_delay is arista.avd.defined %}
+{%                 set poe_link_down_action_cli = poe_link_down_action_cli ~ ' ' ~ ethernet_interface.poe.link_down.power_off_delay %}
+{%             endif %}
+   {{ poe_link_down_action_cli }}
+{%         endif %}
+{%         if ethernet_interface.poe.shutdown.action is arista.avd.defined %}
+   poe shutdown action {{ ethernet_interface.poe.shutdown.action }}
+{%         endif %}
+{%         if ethernet_interface.poe.disabled is arista.avd.defined(true) %}
+   poe disabled
+{%         endif %}
+{%         if ethernet_interface.poe.limit.class is arista.avd.defined %}
+{%             set poe_class_map = dict({0: "15.40", 1: "4.00", 2: "7.00", 3: "15.40", 4: "30.00", 5: "45.00", 6: "60.00", 7: "75.00", 8: "90.00"}) %}
```

Comment:
great way of doing it. But I suggest to use an upper-case variable, to signal that it is a constant, and then set it at the top of the file. Then you avoid setting it for every loop. Also I would except the naked dict would be good enough `set poe_class_map = {0: " ...`